common/src/main/scala/uk/ac/wellcome/sqs/SQSWorker.scala

```diff
@@ -1,47 +1,45 @@
 package uk.ac.wellcome.sqs
 
 import akka.actor.ActorSystem
-import com.amazonaws.services.sqs.model.Message
 import uk.ac.wellcome.utils.JsonUtil._
-import uk.ac.wellcome.exceptions.GracefulFailureException
 import uk.ac.wellcome.metrics.MetricsSender
 import uk.ac.wellcome.models.aws.SQSMessage
-import uk.ac.wellcome.utils.{JsonUtil, TryBackoff}
+import uk.ac.wellcome.utils.JsonUtil.fromJson
 import uk.ac.wellcome.utils.GlobalExecutionContext.context
 import scala.concurrent.Future
-import scala.util.Try
+import com.twitter.inject.Logging
+import scala.concurrent.duration._
 
 abstract class SQSWorker(sqsReader: SQSReader,
                          actorSystem: ActorSystem,
                          metricsSender: MetricsSender)
-    extends TryBackoff {
+    extends Logging {
 
-  lazy val workerName: String = this.getClass.getSimpleName
+  info(s"Starting SQS worker=[$workerName]")
 
-  def processMessage(message: SQSMessage): Future[Unit]
+  lazy val poll = 1 second
+  private lazy val workerName: String = this.getClass.getSimpleName
+  private lazy val scheduler = actorSystem.scheduler
+  private val actor = scheduler.schedule(0 seconds, poll)(processMessages())
 
-  def runSQSWorker(): Unit = run(() => processMessages, actorSystem)
+  def processMessage(message: SQSMessage): Future[Unit]
 
   private def processMessages(): Future[Unit] = {
-    info(s"Starting $workerName.")
-
     sqsReader.retrieveAndDeleteMessages { message =>
-      Future
-        .fromTry(extractMessage(message))
-        .flatMap(m => {
-          debug(s"Processing message: $m")
-
-          metricsSender.timeAndCount(
-            s"${workerName}_ProcessMessage",
-            () => processMessage(m)
-          )
-        })
+      for {
+        m <- Future.fromTry { fromJson[SQSMessage](message.getBody) }
+        _ <- Future.successful { debug(s"Processing message: $m") }
+        metricName = s"${workerName}_ProcessMessage"
+        _ <- metricsSender.timeAndCount(metricName, () => processMessage(m))
+      } yield ()
+    } recover {
+      case _: Throwable => terminalFailureHook()
```

Comment:
Is this correct? If, say Json parsing of the message fails, it should throw a `GracefulFailureException` which is not a terminal failure. I know it will only affect alarms we get and it's not an actual terminal failure (so maybe we should also rename that method)

Reply:
I think the `GracefulFailureException` is handled by the [`SQSReader`](https://github.com/wellcometrust/platform/blob/1d79970cf8084b926b5165405b13e4ba4f4f1781/common/src/main/scala/uk/ac/wellcome/sqs/SQSReader.scala#L72). I think by the time execution to this code we no longer need to worry about it.

Is this correct?

Reply:
oh right, yep you're correct